### DISCUSSION TITLE
Windows 11

### DISCUSSION THREAD

**Discussion by santiagoperdomo** [dictation-toolbox/Caster#939] [Q&A] [answered]
Hi all,

I love Caster; it's saved my job!! Does anyone know if it works with Windows 11? Thank you!!

**Comment by LexiconCode** [comment, marked as answer]
> Hi all,
> 
> I love Caster; it's saved my job!! Does anyone know if it works with Windows 11? Thank you!!

Yes, Castor works with Windows 11.


However, if you're using DPI I would recommend staying with Windows 10 unless you're using DPI v16 as DPI v15 has significant issues with Windows 11.

Thank you for the positive note. I love hearing about people's successes.

**Reply by santiagoperdomo** [reply]
Unfortunately my company is forcing me into a new computer with Windows 11, otherwise I would totally stay with my Windows 10 where Caster has been working for a couple of years. Thanks for the heads up about DPI versions though.

Are there any plans to upgrade castor to use Python 3? I'm having a difficult time installing with Python 2.

For real though, I'm a software developer with three young kids and started having RSI issues two years ago. Caster has allowed me to stay in a career I actually really like and keep providing for my family! I cannot say thank you enough!!!

**Reply by LexiconCode** [reply]
> Are there any plans to upgrade castor to use Python 3? I'm having a difficult time installing with Python 2.

Yes, caster is python 3 compatible and there is a compatible build DPI 13 through 16 natlink with python3.10 32-bit. See the issue pinned https://github.com/dictation-toolbox/Caster/issues/911

**Reply by santiagoperdomo** [reply]
Awesome; thanks for pointing me in the right direction!! So I think my company's firewall isn't letting the install file download the requirements. Is my best bet to get each requirement manually? Will Caster run if I get all the requirements? 

If you have any ideas on how to best go about this, I'd be forever grateful. Thank you everyone for the quick responses!


<img width="855" alt="Screenshot 2023-09-28 214839" src="https://github.com/dictation-toolbox/Caster/assets/38016526/5a914845-ed93-477c-ac98-00cd392ba160">

**Reply by LexiconCode** [reply]
Try `python -m pip install --trusted-host pypi.python.org --trusted-host files.pythonhosted.org --trusted-host pypi.org --upgrade pip`
or 
https://stackoverflow.com/a/14447068/7663566

You night have to install https://pypi.org/project/natlinkcore/ as well before the natlink installer

**Reply by santiagoperdomo** [reply]
@LexiconCode , The stackoverflow link was exactly what I was looking for! Pillow was giving me all kinds of trouble installing saying it needed zlib... I fought with it for a few hours, when I finally stumbled upon https://stackoverflow.com/questions/41188838/cant-install-pillow-in-windows And was able to install it via its Windows installer package found here https://www.lfd.uci.edu/~gohlke/pythonlibs/#pillow.

I have the marvelous caster running on my new computer!! thank you so much for pointing me in the right direction and for all your work to provide this beauty!

**Comment by Azarious** [comment]
I also went through a stretch of several years where I had serious pain
from typing, and Caster was indispensable in doing my job (engineering, but
a fair amount of coding). Thanks to all who have made this possible.

I think I am currently running the DropPy2 build of Caster with Python 3 on
my Windows 10 computer. It looks like that is not being actively developed
though.

**Reply by santiagoperdomo** [reply]
@Azarious , sorry if this is obvious, but what is "DropPy2 build of Caster"? A quick Google search didn't show anything...

**Reply by LexiconCode** [reply]
It's a branch dropping python 2 support. In GitHub there is master branch and DropPy2 branch.

There's further developments made on my own fork if you look at my branches rather than upstream.

**Comment by LexiconCode** [comment]
> It looks like that is not being actively developed though.

My time elsewhere has been sucked up by Natlink. In addition, mid migration Qt quit releasing binaries for 32 bit which means I can't use it with Python 32 bit. So I'm having to migrate GUI code.

Combined with life events and some of the things that's taking much longer than expected.
